prometheus_client/metrics_core.py

```diff
@@ -121,10 +121,12 @@ def __init__(self,
         if name.endswith('_total'):
             name = name[:-6]
         Metric.__init__(self, name, documentation, 'counter', unit)
-        if labels is not None and value is not None:
-            raise ValueError('Can only specify at most one of value and labels.')
+
         if labels is None:
             labels = []
+        elif value is not None:
```

Comment:
I find the old logic to be clearer. It was clearly a check at the very beginning to make sure the input was valid, then we assigned a default value. Now we are doing both at the same time which IMO is harder to understand.